src/Core/Application/Common/Specification/SpecificationBuilderExtensions.cs

```diff
@@ -79,29 +81,184 @@ public static ISpecificationBuilder<T> PaginateBy<T>(this ISpecificationBuilder<
+        if (filter.Filters.Any())
+        {
+            bExpresion = AddFilters(filter.Filters, parameter, bExpresion);
+        }
+
+        return bExpresion;
+    }
+    private static (ConstantExpression CExpresion, ConstantExpression CExpresionAux) GetConstantExpressionFromFilter(MemberExpression mapProperty, Filter filter)
+    {
+        ConstantExpression cExpresionAux = default!;
+
+        ConstantExpression cExpresion;
+        if (mapProperty.Type.IsEnum)
+        {
+            string? stringEnum = GetStringFromJsonElement(filter.Value!);
+
+            if (!Enum.TryParse(mapProperty.Type, stringEnum, true, out object? valueparsed)) throw new CustomException("Value {0} is not valid for {1}");
+
+            cExpresion = Expression.Constant(valueparsed, mapProperty.Type);
+        }
+        else if (mapProperty.Type == typeof(Guid))
+        {
+            string? stringGuid = GetStringFromJsonElement(filter.Value!);
+
+            if (!Guid.TryParse(stringGuid, out Guid valueparsed)) throw new CustomException("Value {0} is not valid for {1}");
+
+            cExpresion = Expression.Constant(valueparsed, mapProperty.Type);
+        }
+        else if (mapProperty.Type == typeof(string))
+        {
+            string? text = GetStringFromJsonElement(filter.Value!);
+
+            cExpresion = Expression.Constant(text, mapProperty.Type);
+        }
+        else
+        {
+            cExpresion = Expression.Constant(Convert.ChangeType(((JsonElement)filter.Value!).GetRawText(), mapProperty.Type), mapProperty.Type);
+            if (filter.ValueAux is not null)
+            {
+                cExpresionAux = Expression.Constant(Convert.ChangeType(((JsonElement)filter.ValueAux).GetRawText(), mapProperty.Type), mapProperty.Type);
+            }
+        }
+
+        return (cExpresion, cExpresionAux);
+    }
+
+    private static BinaryExpression AddFilters(IEnumerable<Filter> filters, ParameterExpression parameter, BinaryExpression bExpresionBase = default!)
+    {
+        BinaryExpression bResult = default!;
+        foreach (Filter filter in filters)
+        {
+            var bExpresionFilter = GetBinaryExpressionFromFilter(filter, parameter);
+
+            if (bExpresionBase is not null)
+            {
+                bResult = filter.Logic switch
+                {
+                    FilterLogic.AND => Expression.And(bExpresionBase, bExpresionFilter),
+                    FilterLogic.OR => Expression.Or(bExpresionBase, bExpresionFilter),
+                    FilterLogic.XOR => Expression.ExclusiveOr(bExpresionBase, bExpresionFilter),
+                    _ => throw new ArgumentException("FilterLogic is not valid.", nameof(FilterLogic)),
+                };
+            }
+            else
+            {
+                bResult = bExpresionFilter;
+            }
+
+        }
+
+        return bResult;
+    }
+
+    public static IOrderedSpecificationBuilder<T> AdvancedFilter<T>(

```

Comment:
Why IOrderedSpecificationBuilder?

Reply:
base on the value returned in -> IOrderedSpecificationBuilder<T> AdvancedSearch<T>

Reply:
well it's wrong there too... only "OrderBy" should return an IOrderedSpecificationBuilder... And I think our OrderBy shouldn't either return an IOrderedSpecificationBuilder, as our OrderBy is used to order by multiple fields at once, so it's not necessary to chain more "ThenBy" or "ThenByDescending" calls on it...